tests/tests_cloud/conftest.py

```diff
@@ -0,0 +1,42 @@
+import importlib
+import os
+import tempfile
+import types
+from pathlib import Path
+
+import pytest
+
+import lightning
+import lightning.store
+
+
+def reload_package(package):
+    # credit: https://stackoverflow.com/a/28516918/4521646
+    assert hasattr(package, "__package__")
+    fn = package.__file__
+    fn_dir = os.path.dirname(fn) + os.sep
+    module_visit = {fn}
+    del fn
+
+    def reload_recursive_ex(module):
+        importlib.reload(module)
+
+        for module_child in vars(module).values():
+            if not isinstance(module_child, types.ModuleType):
+                continue
+            fn_child = getattr(module_child, "__file__", None)
+            if (fn_child is not None) and fn_child.startswith(fn_dir) and fn_child not in module_visit:
+                # print("reloading:", fn_child, "from", module)
+                module_visit.add(fn_child)
+                reload_recursive_ex(module_child)
+
+    return reload_recursive_ex(package)
+
+
+@pytest.fixture(scope="function", autouse=True)
```

Comment:
You probably don't want to do this on every test. It would be sufficient to do it only at the beginning of the test suite.

Reply:
I want as the original cleanup function was also called to each test to prevent using previously downloaded models

Reply:
Yes, it's just highly inefficient.

Reply:
true, but would need to revisit all tests to see if they really need special cleaning

Reply:
so lets get this one in as the living version is really bad :/
and revisit optimization later on... :)